code/processes/wdb.q

```diff
@@ -232,14 +232,26 @@ doreload:{[pt]
 	];
 	};
 
+// set .z.zd to control how data gets compressed
+setcompression:{[compression] if[ 3= count compression;
+							.lg.o[`compression;"setting compression level to (",(";" sv string compression),")"];
+                        				.z.zd:compression;
+                        				.lg.o[`compression;".z.zd has been set to (",(";" sv string .z.zd),")"]]}
+//reset .z.zd to 16 0 0 (i.e no compression)
+removecompression:{[compression] if[ 3= count compression;
+					    		.z.zd:16 0 0;
+                                            		.lg.o[`compression;"resetting compression level to (",(";" sv string 16 0 0),")"]]}
+
+
+
 endofdaysortdate:{[dir;pt;tablist]
 	/-sort permitted tables in database
 	/- sort the table and garbage collect (if enabled)
 	.lg.o[`sort;"starting to sort data"];
 	
 	$[(0 < count .z.pd[]) and ((system "s")<0);
 		[.lg.o[`sort;"sorting on slave sort", string .z.p];
-		{[x] .sort.sorttab[x];if[gc;.gc.run[]]} peach tablist,'.Q.par[dir;pt;] each tablist;
+		{[x] setcompression[compression];.sort.sorttab[x];removecompression[compression];if[gc;.gc.run[]]} peach tablist,'.Q.par[dir;pt;] each tablist;
 		];
```

Comment:
I've fixed that problem now, but I've had to add a compression parameter into a few functions (informsortandreload, endofdaysort, endofdaymerge, endofdaysortdate) so that it feeds through to the sort/sortslave processes. Is this ok or is there a better way to do it ?